plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -808,69 +808,83 @@ private function months_filter() {
 	 *
 	 * @return \stdClass[]
 	 */
-	protected function get_months_filter_options(): array {
+	public function get_months_filter_options(): array {
 		global $wpdb;
 
-		$orders_table = esc_sql( OrdersTableDataStore::get_orders_table_name() );
-		$trash_status = esc_sql( OrderStatus::TRASH );
-
-		$first_year_month_gmt = $wpdb->get_row(
-			// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$orders_table   = esc_sql( OrdersTableDataStore::get_orders_table_name() );
+		$min_max_months = $wpdb->get_row(
 			$wpdb->prepare(
 				"
-					SELECT YEAR( t.date_created_gmt ) AS year,
-					       MONTH( t.date_created_gmt ) AS month
-					FROM $orders_table t
+					SELECT MIN( t.date_created_gmt ) as min_date_gmt,
+					       MAX( t.date_created_gmt ) as max_date_gmt
+					FROM `{$orders_table}` t
 					WHERE type = %s
 					AND status != %s
```

Comment:
@joshuatf do you have thoughts about the performance of this query? Or, do you still have a site available with a high order count that you could use to test this with? If not, I can try to spin something up. (This is a further revision of the work done in #55510)